Python/optimizer_bytecodes.c

```diff
@@ -1023,7 +1028,7 @@ dummy_func(void) {
     }
 
     op(_REPLACE_WITH_TRUE, (value -- res)) {
-        REPLACE_OP(this_instr, _POP_TOP_LOAD_CONST_INLINE_BORROW, 0, (uintptr_t)Py_True);
+        REPLACE_OPCODE_IF_EVALUATES_PURE(value);
```

Comment:
I think this is a pessimization. It used to unconditionally replace the op, now it only does if `value` is a "safe" type.